### DISCUSSION TITLE
How to get stream result on client?

### DISCUSSION THREAD

**Discussion by chengfengfengwang** [honojs/hono#2765] [Q&A]
I expected this code to send two messages: the first one being 'Hello', and the second one 'Hono!' after 3 seconds. However, I only received one message after 3 seconds. I am new to Hono. Can you help me?
```
// app.ts
import { stream, streamText, streamSSE } from "hono/streaming";
import { Hono } from "hono";
import { cors } from "hono/cors";
const app = new Hono();
app.use("*", cors());
app.get("/chat", async (c) => {
  return streamText(c, async (stream) => {
    await stream.writeln('Hello')
    // Wait 1 second.
    await stream.sleep(3000)
    // Write a text without a new line.
    await stream.write(`Hono!`)
  })
});
export default app;
```
```
// client.js
const res =  await fetch('http://localhost:8787/chat');
const reader = res.body.getReader();
while (true) {
  const { done, value } = await reader.read();
  if (done) {
    break;
  }
  const text = new TextDecoder().decode(value);
  console.log("--------"); // after 3s I just received one single log
  console.log(text);
}
```

**Comment by chengfengfengwang** [comment]
This issue has been bothering me for a day. I found that when I develop locally using Wrangler, the messages do not flow. However, when deployed to Cloudflare Worker, the code behaves as I expected! I don't know if this is a Wrangler issue or a Hono issue.

**Reply by AleksandrHovhannisyan** [reply]
Did you ever figure this out? I'm seeing the same thing.

Edit: Figured out what the problem was. I was testing with curl but not passing in the `-N` (no-buffer) flag. So the results were buffering anyway.

**Comment by icealtria** [comment]
https://hono.dev/docs/helpers/streaming#streamtext

If you are developing an application for Cloudflare Workers, a streaming may not work well on Wrangler. If so, add Identity for Content-Encoding header.

```js
app.get('/streamText', (c) => {
  c.header('Content-Encoding', 'Identity')
  return streamText(c, async (stream) => {
    // ...
  })
})
```
